utils/python/CIME/XML/env_mach_specific.py

```diff
@@ -189,7 +194,16 @@ def _compute_actions(self, nodes, child_tag, compiler, debug, mpilib):
                 for child in node:
                     expect(child.tag == child_tag, "Expected %s element" % child_tag)
                     if (self._match_attribs(child.attrib, compiler, debug, mpilib)):
-                        result.append( (child.get("name"), child.text) )
+                        val = child.text
+                        if val is not None:
+                            # We allow a couple special substitutions for these fields
+                            for repl_this, repl_with in [("$COMPILER", compiler), ("$MPILIB", mpilib)]:
+                                val = val.replace(repl_this, repl_with)
+
+                            val = self.get_resolved_value(val)
+                            expect("$" not in val, "Not safe to leave unresolved items in env var value: '%s'" % val)
```

Comment:
I just looked at the mira environment variables settings:
```
    <module_system type="soft">
      <init_path lang="csh">/etc/profile.d/00softenv.csh</init_path>
      <cmd_path lang="csh">soft</cmd_path>
      <init_path lang="sh">/etc/profile.d/00softenv.sh</init_path>
      <cmd_path lang="sh">soft</cmd_path>
      <modules>
        <command name="add">+mpiwrapper-xl</command>
        <command name="add">@ibm-compilers-2015-02</command>
        <command name="add">+cmake</command>
        <command name="add">+python</command>
      </modules>
    </module_system>
    <environment_variables>
      <env name="MPI_TYPE_MAX">10000</env>
      <env name="OMP_DYNAMIC">FALSE</env>
      <env name="OMP_STACKSIZE">64M</env>
      <env name="HDF5">/soft/libraries/hdf5/1.8.14/cnk-xl/current</env>
    </environment_variables>
```
... and it doesn't look like it will be a problem.